crates/nu-command/src/filters/where_.rs

```diff
@@ -10,13 +10,21 @@ impl Command for Where {
     }
 
     fn description(&self) -> &str {
-        "Filter values based on a row condition."
+        "Filter values of an input list based on a condition."
     }
 
     fn extra_description(&self) -> &str {
-        r#"This command works similar to 'filter' but allows extra shorthands for working with
-tables, known as "row conditions". On the other hand, reading the condition from a variable is
-not supported."#
+        r#"A condition is evaluated for each element of the input, and only elements which meet the condition are included in the output.
+
+A condition can be either a "row condition", or a closure. A row condition is a special short-hand syntax to makes accessing fields easier.
+Each element of the input can be accessed through the $it variable.
+
+On the left hand side of a row condition, any field name is automatically expanded to use $it.
+For example, where type == dir is equivalent to where $it.type == dir. This expansion does not happen when passing a subexpression or closure to where.
```

Comment:
I'm not sure whether to suggest that `where type == dir` be wrapped in "something" (perhaps backticks)?  I know that the backticks aren't handled in the help, but I'm not sure that someone *reading* that line that isn't as familiar with the syntax could visually separate the *command* from the surrounding *description*.

Reply:
good point, i tried to avoid it but we honestly have enough command descriptions which do this already that it's probably fine. would be nice if we italicized or colored text in backticks in help text, maybe

Reply:
Was thinking the same thing - Some different offset styling.

Reply:
```suggestion
For example, `where type == dir` is equivalent to `where $it.type == dir`. This expansion does not happen when passing a subexpression or closure to where.
```

I think using backticks here makes sense, it's pretty common in descriptions of other commands:
```nushell
scope commands | select name description extra_description | find -r '`.*?`'
```

Reply:
okay fine, i'll succumb to using the backticks :laughing: we'll just have to finally actually add formatting to `help` text (and maybe errors as well?) with backticks :smile: